packages/flutter/lib/src/cupertino/theme.dart

```diff
@@ -0,0 +1,314 @@
+  }) : assert(child != null),
+       assert(data != null),
+       super(key: key, child: child);
+
+  /// The [CupertinoThemeData] styling for this theme.
+  final CupertinoThemeData data;
+
+  @override
+  bool updateShouldNotify(CupertinoTheme oldWidget) => data != oldWidget.data;
+
+  /// Retrieve the [CupertinoThemeData] from an ancestor [CupertinoTheme] widget.
+  ///
+  /// May return a default [CupertinoThemeData] if no [CupertinoTheme] widgets
+  /// exist in the ancestry tree.
+  ///
+  /// No [InheritedModel] dependency is created when the [CupertinoThemeData]
+  /// object is provided but a dependency is created when properties on the
+  /// [CupertinoThemeData] are read. When a dependency is created, a rebuild
+  /// is invoked on the reader when the read properties are changed.
+  static CupertinoThemeData of(BuildContext context) {
+    final CupertinoTheme theme = context.inheritFromWidgetOfExactType(CupertinoTheme);
+    return theme?.data ?? const CupertinoThemeData();
+  }
+}
+
+/// Styling specifications for a [CupertinoTheme].
+///
+/// All constructor parameters can be null, in which case a
+/// [CupertinoColors.activeBlue] based default iOS theme styling is used.
+///
+/// Parameters can also be partially specified, in which case some parameters
+/// will cascade down to other dependent parameters to create a cohesive
+/// visual effect. For instance, if a [primaryColor] is specified, it would
+/// cascade down to affect some fonts in [textTheme] if [textTheme] is not
+/// specified.
+///
+/// When retrieved using [CupertinoTheme.of], reading each property will create
+/// a dependency in the [BuildContext] given to [CupertinoTheme.of]. The
+/// given [BuildContext] will rebuild when that property changes.
+///
+/// See also:
+///
+///  * [CupertinoTheme], in which this [CupertinoThemeData] is inserted.
+///  * [ThemeData], a Material equivalent that also configures Cupertino
+///    styling via a [CupertinoThemeData] subclass [MaterialBasedCupertinoThemeData].
+@immutable
+class CupertinoThemeData extends Diagnosticable {
+  /// Create a [CupertinoTheme] styling specification.
+  ///
+  /// Unspecified parameters default to a reasonable iOS default style.
+  const CupertinoThemeData({
+    Brightness brightness,
+    Color primaryColor,
+    Color primaryContrastingColor,
+    CupertinoTextTheme textTheme,
+    Color barBackgroundColor,
+    Color scaffoldBackgroundColor,
+    Color tableBackgroundColor,
+  }) : this.raw(
+        brightness,
+        primaryColor,
+        primaryContrastingColor,
+        textTheme,
+        barBackgroundColor,
+        scaffoldBackgroundColor,
+        tableBackgroundColor,
+      );
+
+  /// A raw constructor used by subclasses to populate the superclass's fields
+  /// to get superclass behaviors not overriden by subclasses.
+  ///
+  /// Same as the default constructor but with positional arguments to avoid
+  /// forgetting any.
+  @protected
+  const CupertinoThemeData.raw(
```

Comment:
do we actually need this? we only need it in ThemeData because the default constructor does all kinds of magic.